### ISSUE TITLE
Add on option to output a checkpoint every x minutes

### ISSUE THREAD

**Issue by wulu473** [huggingface/transformers#13830]
# 🚀 Feature request

Extending `transformers.trainer_utils.IntervalStrategy` by an option `MINUTES` to be able to save a checkpoint based on training time.

## Motivation

For hyper-parameter searches over large batch sizes (where gradient accumulation is necessary) the time between steps can vary by orders of magnitude between different runs. This could lead to very different number of written checkpoints for different runs. In this case it could be useful to specify to have checkpoints output every x hours rather than every x steps. 

## Contribution

Happy to submit a PR for that.

**Comment by LysandreJik** [comment]
cc @sgugger

**Comment by sgugger** [comment]
This sounds like a useful feature, feel free to suggest a PR to add that!

**Comment by sgugger** [comment]
Unstale

**Comment by sgugger** [comment]
Unstale
